routing/single_mwm_router.cpp

```diff
@@ -255,17 +230,28 @@ bool SingleMwmRouter::RedressRoute(MwmSet::MwmId const & mwmId, IVehicleModel co
 
 // static
 unique_ptr<SingleMwmRouter> SingleMwmRouter::CreateCarRouter(
-    Index const & index, traffic::TrafficCache const & trafficCache)
+    Index & index, TCountryFileFn const & countryFileFn, shared_ptr<NumMwmIds> numMwmIds,
+    traffic::TrafficCache const & trafficCache)
 {
   auto vehicleModelFactory = make_shared<CarModelFactory>();
   // @TODO Bicycle turn generation engine is used now. It's ok for the time being.
   // But later a special car turn generation engine should be implemented.
-  auto directionsEngine = make_unique<BicycleDirectionsEngine>(index);
-  auto estimator =
-      EdgeEstimator::CreateForCar(*vehicleModelFactory->GetVehicleModel(), trafficCache);
-  auto router =
-      make_unique<SingleMwmRouter>("astar-bidirectional-car", index, trafficCache,
-                                   move(vehicleModelFactory), estimator, move(directionsEngine));
+  auto directionsEngine = make_unique<BicycleDirectionsEngine>(index, numMwmIds);
+
+  double maxSpeed = 0.0;
+  numMwmIds->ForEachId([&](NumMwmId id) {
```

Comment:
Do you really want to iterate over all country files? Or just over country files that are present on device?